### ISSUE TITLE
AssetStudio crashes on Dreamfall Chapters

### ISSUE THREAD

**Issue by creibens** [Perfare/AssetStudio#263]
I recently installed the game Dreamfall Chapters on Steam. When I try do load its folder, it starts working normally, but after a while, it crashes without any error message. The last message I saw was something like "Loading CAB-xxxxx.resource", where "xxxxx" is a lengthy sequence of obviously hexadecimal numbers. So I tried to isolate that file, but I couldn't find a file with such a name.

My OS is Windows 10 pro 64 bit version 1803, build 17134.228, german. The event log shows this (sorry, german):

Name der fehlerhaften Anwendung: AssetStudio.exe, Version: 0.10.0.83, Zeitstempel: 0x5b960544
Name des fehlerhaften Moduls: KERNELBASE.dll, Version: 10.0.17134.165, Zeitstempel: 0xb0bb231d
Ausnahmecode: 0xe0434352
Fehleroffset: 0x000000000003a388
ID des fehlerhaften Prozesses: 0x29bc
Startzeit der fehlerhaften Anwendung: 0x01d449c7d29a1759
Pfad der fehlerhaften Anwendung: C:\Program Files\AssetStudio\AssetStudio.exe
Pfad des fehlerhaften Moduls: C:\WINDOWS\System32\KERNELBASE.dll
Berichtskennung: a423a0a1-cddd-480b-88e1-104b9b5004a4
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist: 

HTH
Claus

**Comment by cesm1980** [comment]
I can confirm i have the exact same problem with that game (altough the package that makes it crash seems to be random) and Life Is Strange: Before the Storm as well, after some testings i noticed that assetstudio starts using HUGE amounts of memory (7-10gb) in certain packages EVEN if they are just 400-500mg (can't remember exactly now) so apparently these specific games assets create some kind of memory leak, it doesn't appear to seem related to the total size of assets because there's other games that are loaded without any problems and are bigger than these 2 ones.

By the way creibens and Perfare, both games extract models and animations without problems, even tough i have to load specific assets only to avoid that crash.

**Comment by HeresHavi** [comment]
Well, this is one of the reasons why it's not a good idea to load assets over 2 GB. Prefare has mentioned this before and then he allowed AssetStudio to load over 2 GB and included some fixes. This could be part of it. So, maybe do it in parts.

**Comment by Perfare** [comment]
Need a game file so I can check the cause of the error.

**Comment by cesm1980** [comment]
Ok done it! Found what causes the problem, and i think it's the same files that the op was complaning about since it's the only place where i found CAB files decompressing with weird long names : there's a subfolder on the DATA folder called : Audio\Bundles that contains the sounds of the game in german(1gb) and english(900mb). Assetstudio crashes if i try to load JUST this audio folder even without the big main game assets BUT it doesn't crash anymore if i remove the german or the english languages, so the problem is assetstudio tries to load both folders at the same time, even without the other game files, also i notice every single audio is decompressed which makes the memory usage increase more than usual.

Then i tried just the normal game asset files (which is 18gb!!) without the Audio folder and even tough it took quite a few minutes to load everything and used around 4gb of memory it did NOT crash asset studio! (because i moved the audio folder somewhere else)

Not sure if the OP is interested in the game sounds or not, but in case he is, you perfare still want the audio files just in case? (nearly 2gb)

**Comment by creibens** [comment]
Thanks for your help. I now tried to load the data folder in three steps:

1. Loading subfolder Audio\Bundles\de separately
2. Loading subfolder Audio\Bundles\en separately
3. Loading all other subfolders in one go

That did it. Thank you.

And yes, it's the game sounds I am interested in ;-)

**Comment by cesm1980** [comment]
Yeah, you were lucky i also tested the same game very recently, and i indeed noticed the same problem you had, so it was just a matter of me doing a little testing.
